File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
input = sys.stdin.readline
sys.setrecursionlimit(10**7)


def dfs(v):
    for tmp in graph[v]:
        if not visited[tmp]:
            visited[tmp] = True
            print(tmp, end=' ')
            dfs(tmp)


n, m, v = map(int, input().split())
graph = [[] for _ in range(n+1)]

for i in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for i in range(1, n+1):
    graph[i].sort()
visited = [False] * (n+1)
visited[v] = True
print(v, end=' ')
dfs(v)
print()

visited = [False] * (n+1)
bfsl = deque()
bfsl.append(v)
visited[v] = True
while True:
    if not bfsl:
        break
    tmp = bfsl.popleft()
    print(tmp, end=' ')

    for i in graph[tmp]:
        if not visited[i]:
            visited[i] = True
            bfsl.append(i)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;

public class Main {

    static List<Integer>[] graph;
    static boolean[] visited;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt(); // 노드의 개수
        int m = sc.nextInt(); // 간선의 개수
        int v = sc.nextInt(); // 시작 노드

        graph = new ArrayList[n + 1];
        for (int i = 1; i <= n; i++) {
            graph[i] = new ArrayList<>();
        }

        for (int i = 0; i < m; i++) {
            int a = sc.nextInt();
            int b = sc.nextInt();
            graph[a].add(b);
            graph[b].add(a);
        }

        for (int i = 1; i <= n; i++) {
            Collections.sort(graph[i]);
        }

        visited = new boolean[n + 1];
        visited[v] = true;
        System.out.print(v + " ");
        dfs(v);
        System.out.println();

        visited = new boolean[n + 1];
        Queue<Integer> bfsl = new LinkedList<>();
        bfsl.add(v);
        visited[v] = true;

        while (!bfsl.isEmpty()) {
            int tmp = bfsl.poll();
            System.out.print(tmp + " ");

            for (int i : graph[tmp]) {
                if (!visited[i]) {
                    visited[i] = true;
                    bfsl.add(i);
                }
            }
        }

        sc.close();
    }

    static void dfs(int v) {
        for (int tmp : graph[v]) {
            if (!visited[tmp]) {
                visited[tmp] = true;
                System.out.print(tmp + " ");
                dfs(tmp);
            }
        }
    }
}
